Sources/rapidsnark/rapidsnark.swift

```diff
@@ -3,201 +3,198 @@
+    var currentPublicBufferSize : Int;
     if let publicBufferSize {
-        public_buffer_size = publicBufferSize;
+        currentPublicBufferSize = publicBufferSize;
     } else {
-        public_buffer_size = try! groth16PublicSizeForZkeyBuf(zkey: zkey, errorBufferSize: errorBufferSize);
+        do{
+            currentPublicBufferSize = try groth16PublicSizeForZkeyBuf(zkey: zkey, errorBufferSize: errorBufferSize);
+        }catch{
+            throw error
+        }
     }
 
-    var proof_buffer_size = UInt(proofBufferSize);
     var proofBuffer = Array<CChar>(repeating: 0, count: proofBufferSize);
-
-    var public_buffer_size_uint = UInt(public_buffer_size)
-    var publicBuffer = Array<CChar>(repeating: 0, count: public_buffer_size)
-
-    let error_buffer_size = UInt(errorBufferSize)
+    var publicBuffer = Array<CChar>(repeating: 0, count: currentPublicBufferSize)
     var errorMessageBuffer: [CChar] = Array(repeating: 0, count: errorBufferSize)
+    
+    var proofBufferSizeUInt = UInt(proofBufferSize)
+    var currentPublicBufferSizeUInt = UInt(currentPublicBufferSize)
+    var errorBufferSizeUInt = UInt(errorBufferSize)
 
+    // Call the rapidsnark C++ library function to perform the Groth16 proof
     let statusCode = groth16_prover(
         zkeyBuf, UInt(zkey.count),
         witnessBuf, UInt(witness.count),
-        &proofBuffer, &proof_buffer_size,
-        &publicBuffer, &public_buffer_size_uint,
-        &errorMessageBuffer, error_buffer_size
+        &proofBuffer, &proofBufferSizeUInt,
+        &publicBuffer, &currentPublicBufferSizeUInt,
+        &errorMessageBuffer, errorBufferSizeUInt
     );
 
     if (statusCode == PROVER_OK) {
-        let proof = String(cString: proofBuffer)
-        let publicSignals = String(cString: publicBuffer)
-
-        return (proof, publicSignals)
+        return (String(cString: proofBuffer), String(cString: publicBuffer))
     }
 
-    let error = String(cString: errorMessageBuffer)
-
-    if (statusCode == PROVER_ERROR) {
-        throw RapidsnarkProverError.error(message: error);
-    } else if (statusCode == PROVER_ERROR_SHORT_BUFFER) {
-        throw RapidsnarkProverError.shortBuffer(message: error);
-    } else if (statusCode == PROVER_INVALID_WITNESS_LENGTH) {
-        throw RapidsnarkProverError.invalidWitnessLength(message: error);
-    }
-
-    throw RapidsnarkUnknownStatusError()
+    throw groth16proverStatusCodeErrors(statusCode, message: String(cString: errorMessageBuffer))
 }
 
+// Performs cryptographic proofs using the Groth16 proving scheme
 public func groth16ProveWithZKeyFilePath(
-    zkeyPath: String,
+    zkeyPath: String, // Path to .zkey file
     witness: Data,
-    proofBufferSize: Int = DEFAULT_PROOF_BUFFER_SIZE,
-    publicBufferSize: Int? = nil,
-    errorBufferSize: Int = DEFAULT_ERROR_BUFFER_SIZE
+    proofBufferSize: Int = defaultProofBufferSize, // Optional: Sets the size of the buffer to store the proof, with a default size
```

Comment:
These can be put into documentation comments, which I think you were going to do.
https://nshipster.com/swift-documentation/

Reply:
Nice, suggestion, did that